packages/models-library/src/models_library/service_settings.py

```diff
@@ -59,11 +66,145 @@ class Config:
         }
 
 
-class ServiceSettings(BaseModel):
-    __root__: List[ServiceSetting]
+class SimcoreServiceSettings(BaseModel):
+    __root__: List[SimcoreServiceSetting]
 
     def __iter__(self):
         return iter(self.__root__)
 
     def __getitem__(self, item):
         return self.__root__[item]
+
+
+class PathsMapping(BaseModel):
+    inputs_path: Path = Field(
+        ..., description="path where the service expects all the inputs folder"
+    )
+    outputs_path: Path = Field(
+        ..., description="path where the service expects all the outputs folder"
+    )
+    other_paths: List[Path] = Field(
+        [],
+        description="optional list of path which contents need to be saved and restored",
+    )
+
+    @validator("other_paths", always=True)
+    @classmethod
+    def convert_none_to_empty_list(cls, v):
+        return [] if v is None else v
+
+    class Config(_BaseConfig):
+        schema_extra = {
+            "examples": {
+                "outputs_path": "/tmp/outputs",  # nosec
+                "inputs_path": "/tmp/inputs",  # nosec
+            }
+        }
+
+
+ComposeSpecModel = Optional[Dict[str, Any]]
+
+
+class SimcoreService(BaseModel):
```

Comment:
what about SimcoreLabels? or SimcoreServiceLabels?

Reply:
I've renamed almost all the entries in this module.